pkg/services/ngalert/notifier/channels/googlechat.go

```diff
@@ -98,21 +99,25 @@ func (gcn *GoogleChatNotifier) Notify(ctx context.Context, as ...*types.Alert) (
 	}
 
 	ruleURL := joinUrlPath(gcn.tmpl.ExternalURL.String(), "/alerting/list", gcn.log)
```

Comment:
I suppose this right here might be the underlying issue. Presumably `gcn.tmpl.ExternalURL` is `""` if the template doesn't have any way to know the instance's external URL (missing from the config file, etc.)

I still like this approach of handling it, though - since we have no idea what was actually entered in the config file, we are covering all bases. It doesn't seem sufficient to only check for emptiness here.

I commented suggestions to the log message, to point users toward the config file as a likely culprit.

Nothing to do here, just thought I'd point this out.